server/src/main/java/org/elasticsearch/action/explain/ExplainResponse.java

```diff
@@ -129,4 +157,92 @@ public void writeTo(StreamOutput out) throws IOException {
             getResult.writeTo(out);
         }
     }
+
+    private static final ConstructingObjectParser<ExplainResponse, Void> PARSER = new ConstructingObjectParser<>("explain", true,
+        arg -> new ExplainResponse((String) arg[0], (String) arg[1], (String) arg[2], (boolean) arg[3], (Explanation) arg[4],
+            (GetResult) arg[5]));
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), _INDEX);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), _TYPE);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), _ID);
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), MATCHED);
```

Comment:
I think that you end up passing matched in as exists in the ExplainResponse constructor.